skunk/rust/skunk/src/ir_gen/codegen/c_functions.rs

```diff
@@ -63,6 +63,14 @@ pub fn get_vsnprintf<'ctx>(cg: &mut CodegenState<'ctx>) -> FunctionValue<'ctx> {
   }).unwrap()
 }
 
+pub fn printf<'ctx>(cg: &mut CodegenState<'ctx>, fmt: &str, args: Vec<BasicValueEnum<'ctx>>) {
+  let printf = get_printf(cg);
+  let fmt_str = cg.global_string(fmt);
+  let mut arg_array = vec![fmt_str.into()];
+  arg_array.extend_from_slice(&args[..]);
+  cg.builder.build_call(printf, &arg_array[..], "_");
+}
+
 pub fn get_printf<'ctx>(cg: &CodegenState<'ctx>) -> FunctionValue<'ctx> {
```

Comment:
These could be written in the form:

```impl <'ctx> CodegenState<'ctx> {
 pub fn get_printf(self: Self) -> FunctionValue<'ctx> { ... }
 ```
 
 Which would allow `cg.get_printf()` and would group code a bit more?

Reply:
Yup, that would be better. Filed as our first ever issue: #6 !